runtime/compiler/CMakeLists.txt

```diff
@@ -328,7 +328,7 @@ omr_stringify(J9_CXXFLAGS_STR ${J9_SHAREDFLAGS} ${J9_CXXFLAGS})
 set(CMAKE_CXX_FLAGS "${J9_CXXFLAGS_STR}")
 set(CMAKE_C_FLAGS "${J9_CFLAGS_STR}")
 
-if(OMR_ARCH_X86 OR OMR_ARCH_S390 OR OMR_ARCH_AARCH64)
+if(OMR_ARCH_X86 OR OMR_ARCH_S390 OR OMR_ARCH_AARCH64 OR (OMR_ARCH_POWER AND (__clang__ OR __GNUC__ OR __GNUG__)))
```

Comment:
Suggest this should use `OMR_TOOLCONFIG` instead of `__clang__`, `__GNUC__`, or `__GNUG__`.

Reply:
I see, perhaps `NOT OMR_TOOLCONFIG STREQUAL "xlc"`?

Reply:
I'll let you know once I find out!

Reply:
Some of those should be easy to fix:
* the three flavours of "file/unit is empty" can be avoided by changing the cmake configuration to avoid compiling such files
* the "maximum diagnostics" should fix itself once we fix the others
* `/*` in a comment - trivial

The others should be treated as errors, to me they signal broken code (not just warnings).